adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -301,17 +302,30 @@ final void handleEndpointConnection(final MqttEndpoint endpoint) {
         }
     }
 
-    private void handleConnectionRequestResult ( final MqttEndpoint endpoint, final AsyncResult<Device> done) {
+    private void handleConnectionRequestResult(final MqttEndpoint endpoint,
+            final AsyncResult<Device> authenticationAttempt) {
 
-        if ( done.succeeded() ) {
+        if (authenticationAttempt.succeeded()) {
 
-            endpoint.accept(false);
-            sendConnectedEvent(endpoint.clientIdentifier(), done.result());
+            sendConnectedEvent(endpoint.clientIdentifier(), authenticationAttempt.result())
+                    .setHandler(sendAttempt -> {
+                        if (sendAttempt.succeeded()) {
+                            endpoint.accept(false);
+                        } else {
+                            LOG.warn(
+                                    "connection request from client [clientId: {}] rejected due to connection event "
+                                            + "failure: {}",
+                                    endpoint.clientIdentifier(),
+                                    MqttConnectReturnCode.CONNECTION_REFUSED_SERVER_UNAVAILABLE,
+                                    sendAttempt.cause());
+                            endpoint.reject(MqttConnectReturnCode.CONNECTION_REFUSED_SERVER_UNAVAILABLE);
+                        }
+                    });
 
         } else {
 
-            final Throwable t = done.cause();
-            if ( t instanceof MqttConnectionException ) {
+            final Throwable t = authenticationAttempt.cause();
+            if (t instanceof MqttConnectionException) {
```

Comment:
what if t is not an instance of `MqttConnectionException`?

Reply:
Then it will go into the `else` branch, which will call:

~~~java
endpoint.reject(MqttConnectReturnCode.CONNECTION_REFUSED_SERVER_UNAVAILABLE);
~~~